packages/iov-core/src/workers/signingservice.worker.ts

```diff
@@ -108,82 +32,7 @@ async function main(): Promise<void> {
   // ganache second identity
   await profile.createIdentity(secp256k1Wallet.id, ethereumChainId, HdPaths.bip44(60, 0, 0, 1));
 
-  const serverCore = new ServerCore(profile, signer);
-
-  async function handleRequest(event: any): Promise<JsonRpcResponse | JsonRpcErrorResponse> {
-    let request: JsonRpcRequest;
-    try {
-      request = parseJsonRpcRequest(event.data);
-    } catch (error) {
-      const requestId = parseJsonRpcId(event.data);
-      const errorResponse: JsonRpcErrorResponse = {
-        jsonrpc: "2.0",
-        id: requestId,
-        error: {
-          code: jsonRpcCodeInvalidRequest,
-          message: error.toString(),
-        },
-      };
-      return errorResponse;
-    }
-
-    let call: RpcCall;
-    try {
-      call = parseRpcCall(request);
-    } catch (error) {
-      const errorResponse: JsonRpcErrorResponse = {
-        jsonrpc: "2.0",
-        id: request.id,
-        error: {
-          code: jsonRpcCodeMethodNotFound,
-          message: error.toString(),
-        },
-      };
-      return errorResponse;
-    }
-
-    try {
-      let response: JsonRpcResponse;
-      switch (call.name) {
-        case "getIdentities":
-          response = {
-            jsonrpc: "2.0",
-            id: request.id,
-            result: await serverCore.getIdentities(call.reason, call.chainIds),
-          };
-          break;
-        case "signAndPost":
-          response = {
-            jsonrpc: "2.0",
-            id: request.id,
-            result: await serverCore.signAndPost(call.reason, call.transaction),
-          };
-          break;
-        default:
-          throw new Error("Unsupported RPC call");
-      }
-      return response;
-    } catch (error) {
-      let errorCode: number;
-      if (error instanceof ParamsError) {
-        errorCode = jsonRpcCodeInvalidParams;
-      } else if (error instanceof MethodNotFoundError) {
-        errorCode = jsonRpcCodeMethodNotFound;
-      } else {
-        errorCode = jsonRpcCodeServerErrorDefault;
-      }
-
-      const errorResponse: JsonRpcErrorResponse = {
-        jsonrpc: "2.0",
-        id: request.id,
-        error: {
-          code: errorCode,
-          message: error.toString(),
-        },
-      };
-      return errorResponse;
-    }
-  }
+  const server = new JsonRpcSigningServer(new SigningServerCore(profile, signer));
```

Comment:
I like this example. All logic is gone, just configuration, as should be in a test case :+1: